addons/marketing_card/models/ir_model.py

```diff
@@ -0,0 +1,13 @@
+from odoo import models
+
+
+class IrModel(models.Model):
+    _inherit = 'ir.model'
+
+    def unlink(self):
+        """Remove campaigns on removed models."""
+        self.env['card.campaign'].search([
+            ('res_model', 'in', self.mapped('model'))
+        ]).unlink()
```

Comment:
I see in other modules that it may be done in an ondelete function. To be honest, I am not sure what is the best here nor certain why, when reading commit messages. It seems rather touchy, so probably fair to ask tde / xmo on that part, as apparently columns are deleted first, then the tables, not sure if we can do the search here as expected.